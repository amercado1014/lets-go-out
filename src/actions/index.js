export const addRestaurants = restaurants => ({
  type: 'ADD_RESTAURANTS',
  restaurants
});

export const addMenu = menu => ({
  type: 'ADD_MENU',
  menu
});

export const addRestaurant = restaurant => ({
  type: "ADD_RESTAURANT",
  restaurant
});

export const locationOff = boolean => ({
  type: 'LOCATION_OFF',
  boolean
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
});

export const removeFavorite = apiKey => ({
  type: 'REMOVE_FAVORITE',
  apiKey
});

export const resetFavorites = () => ({
  type: 'RESET_FAVORITES'
});