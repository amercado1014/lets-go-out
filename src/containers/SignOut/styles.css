.signout-button {
  text-decoration: none;
  font-size: 20px;
  color: black;
  background: none;
  margin-right: 10px;
  cursor: pointer;
}

.signout-button:hover {
  color: white;
}