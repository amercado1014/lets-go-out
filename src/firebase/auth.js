import { auth } from "./firebase";

// Sign Up
export const signUp = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

// Sign In
export const signIn = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

// Sign out
export const signOut = () => auth.signOut();

// Password Reset
export const passwordReset = (email) =>
  auth.sendPasswordResetEmail(email);

// Password Change
export const passwordChange = (password) =>
  auth.currentUser.updatePassword(password);