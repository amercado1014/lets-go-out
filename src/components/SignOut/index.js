import React from 'react';
import './styles.css';
import { auth } from '../../firebase';

const SignOut = () => (
  <button type='button' onClick={auth.signOut}>
    Sign Out
  </button>
);

export default SignOut;
