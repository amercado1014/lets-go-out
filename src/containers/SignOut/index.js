import React, { Component} from 'react';
import './styles.css';
import { auth } from '../../firebase';
import { connect } from 'react-redux';
import { resetFavorites } from '../../actions';
import PropTypes from "prop-types";

export class SignOut extends Component {

  handleClick = () => {
    const { resetFavorites } = this.props;
    auth.signOut;
    resetFavorites();
  }

  render () {
    return (
      <button type='button' onClick={this.handleClick}>
        Sign Out
      </button>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  resetFavorites: () => dispatch(resetFavorites())
});

SignOut.propTypes = {
  resetFavorites: PropTypes.func
};

export default connect(null, mapDispatchToProps)(SignOut);
