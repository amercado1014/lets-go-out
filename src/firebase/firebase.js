import * as firebase from "firebase";

const config = {
  apiKey: "AIzaSyBcdjsk8ijIbqaA5RwkZI-Fga0eg4wN-Eo",
  authDomain: "let-s-order-out.firebaseapp.com",
  databaseURL: "https://let-s-order-out.firebaseio.com",
  projectId: "let-s-order-out",
  storageBucket: "let-s-order-out.appspot.com",
  messagingSenderId: "854964870347"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

export { auth };
