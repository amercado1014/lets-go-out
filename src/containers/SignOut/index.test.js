import React from 'react';
import { SignOut, mapDispatchToProps } from '../SignOut';
import { shallow } from 'enzyme';
import { auth } from '../../firebase';
import * as actions from '../../actions';

describe('SignOut', () => {
  let wrapper;
  let mockResetFavorites;

  beforeEach(() => {
    mockResetFavorites = jest.fn();
    wrapper = shallow(<SignOut 
      resetFavorites={mockResetFavorites}/>);
  });

  it('should math the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call auth.signOut on handleClick', () => {
    auth.signOut = jest.fn();
    wrapper.instance().handleClick();
    expect(auth.signOut).toHaveBeenCalled();
  });

  it('should call resetFavorites on handleClick', () => {
    wrapper.instance().handleClick();
    expect(mockResetFavorites).toHaveBeenCalled();
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch on resetFavorites', () => {
    const mockDispatch = jest.fn();
    const expected = actions.resetFavorites();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.resetFavorites();
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
