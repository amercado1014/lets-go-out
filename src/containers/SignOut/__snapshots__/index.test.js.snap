// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut should math the snapshot 1`] = `undefined`;
